hw/ip/aon_timer/doc/_index.md

```diff
@@ -0,0 +1,102 @@
+Both the wakeup and the IRQ signals remain asserted until system reset or explicit acknowledgement by software.
+This first threshold is known as the watchdog bark.
+
+When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which will reset the whole system.
+This is independent of the IRQ sent as part of the watchdog bark.
+The system reset also resets the always-on timer, so software is not required to directly acknowledge anything after a watchdog reset.
+
+To prevent the watchdog bark or bite, software is expected to periodically reset the count when operating normally.
+This is referred to as petting the watchdog, and is achieved by resetting the count to zero.
+
+Since this timer functions as a watchdog, it has two additional functions not present in the always-on wakeup timer:
+* Watchdog configuration lock
+* Watchdog pause in sleep
+
+Unlike the wakeup timer, the watchdog timer configuration might need to be locked by firmware until the next system reset.
+This allows the option of preventing firmware from accidentally or maliciously disabling the watchdog.
+
+The "pause in sleep" option controls whether the watchdog timer continues to count in low-power mode.
+This allows configurations where the watchdog timer can remain programmed and locked while the device is put to sleep for relatively long periods, controlled by the wakeup timer.
+Without this feature, the watchdog timer might wake up the core prematurely by triggering a watchdog bark.
+
+# Theory of Operations
+
+## Block Diagram
+
+![AON Timer Block Diagram](aon_timer_block_diagram.svg)
+
+See the block diagram for high level function and connectivity.
+The timer interacts with the CPU core and the power manager and reset manager to drive wakeup / reset events and interrupts.
+There is also an extra input to tell the counter whether to run ("counter-run").
+This is used to stop the watchdog timer running when in debugging mode or when the alert handler has put the system in a "killed" state.
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/aon_timer/data/aon_timer.hjson" >}}
+
+## Design Details
+
+The always-on timer will run on a ~200KHz clock.
+The timers themselves are 32b wide, giving a maximum timeout window of roughly ~6 hours.
+
+Since the timer core runs on a slow clock, register values are sampled into the main clock domain to ensure register read / writes do not incur large latencies.
+
+# Programmers Guide
+
+## Initialization
+
+1. Write the timer values {{<regref"WKUP_COUNT">}} and {{<regref"WDOG_COUNT">}} to zero.
+2. Program the desired thresholds in {{<regref"WKUP_THOLD">}}, {{<regref"WDOG_BARK_THOLD">}} and {{<regref"WDOG_BITE_THOLD">}}.
+3. Set the enable bit to 1 in the {{<regref"WKUP_CTRL">}} / {{<regref"WDOG_CTRL">}} registers.
+4. If desired, lock the watchdog configuration by writing 1 to the `regwen` bit in {{<regref"WDOG_REGWEN">}}.
+
+## Watchdog pet
```

Comment:
How about the wake-up timer, does it have to be cleared, or is it cleared automatically?

Reply:
Not sure what you mean here - the whole purpose of the timer is to run in low-power and wake up the system.

Reply:
Ah maybe you mean: once it's triggered and woken the system up it will keep on counting (and triggering interrupts). I'll add a hint to the interrupt handling section to disable or reinitialize the counter once it has triggered.

Reply:
> Ah maybe you mean: once it's triggered and woken the system up it will keep on counting (and triggering interrupts).

Yes. Didn't quite explain myself well in the previous comment.